catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy

```diff
@@ -0,0 +1,283 @@
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
+        tmpHomeDir.list() == [] // dir is empty
+    }
+
+    def "Test single metacard no content export"() {
+        setup:
+        exportCommand.with {
+            it.delete = false
+        }
+
+        def attributes = simpleAttributes()
+        attributes.remove(Metacard.RESOURCE_URI) // removed Resource URI simulates no content
+        def result = new ResultImpl(simpleMetacard(attributes))
+        exportCommand.catalogFramework = Mock(CatalogFramework) {
+            query(_ as QueryRequest) >> { QueryRequest req ->
+                new QueryResponseImpl(req, [result], 1)
+            }
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
+        tmpHomeDir.list().size() == 1
+
+        String zip = tmpHomeDir.listFiles().find()?.canonicalPath
+        assert zip?.trim() as boolean // null or empty
+        assert zip.endsWith('.zip')
+
+        def files = new ZipFile(zip).entries()
+                .collect { it.isDirectory() ? null : it.name }
+                .findAll { it != null }
+
+        assert [result.metacard.id].every { id ->
+            files.any { it.contains(id) }
+        }
+
+
+    }
+
+    def "Test single metacard with content export"() {
+        setup:
+        exportCommand.with {
+            it.delete = false
+        }
+
+        def result = new ResultImpl(simpleMetacard(simpleAttributes() + [(Metacard.TAGS): [Metacard.DEFAULT_TAG]]))
+        def resourceName = "contentfor-${result.metacard.id}.xml" as String
+        exportCommand.catalogFramework = Mock(CatalogFramework) {
+            query(_ as QueryRequest) >> { QueryRequest req ->
+                new QueryResponseImpl(req, [result], 1)
+            }
+            getLocalResource(_ as ResourceRequest) >> { ResourceRequest req ->
+                BinaryContent xmlContent = getMetacardToXmlTransformer()(result.metacard, [:])
+
+                return new ResourceResponseImpl(req, [:], new ResourceImpl(xmlContent.inputStream,
+                        new MimeType('text/xml'),
+                        resourceName))
+            }
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
+        tmpHomeDir.list().size() == 1
+        tmpHomeDir.list().find()?.endsWith('.zip')
+
+        String zip = tmpHomeDir.listFiles().find()?.canonicalPath
+        assert zip?.trim() as boolean // null or empty
+        assert zip.endsWith('.zip')
+
+        def files = new ZipFile(zip).entries()
+                .collect { it.isDirectory() ? null : it.name }
+                .findAll { it != null }
+        println(files)
+        assert [result.metacard.id].every { id ->
+            files.any { it.contains(id) }
+        }
+        assert [resourceName].every { name ->
+            files.any { it.contains(name) }
+        }
+
+    }
+
```

Comment:
- added missing xformer test
- Cannot produce null parent directory
- added blank filename test
- added test to ensure deletions are called